src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -634,6 +639,33 @@ void RollBackMsiInstall(WorkloadDownload msiToRollback, DirectoryPath? offlineCa
             }
         }
 
+        public IEnumerable<WorkloadHistoryRecord> GetWorkloadHistoryRecords(string sdkFeatureBand)
```

Comment:
This looks like the same code as in `FileBasedInstaller`, perhaps we should create a helper method somewhere that both installers call to read the history records from a directory.